Makefile

```diff
@@ -1,110 +1,90 @@
 .PHONY: setup
 
-sweep:
-	@echo "WARNING: This will destroy infrastructure. Use only in development accounts."
-	SNOWFLAKE_ENABLE_SWEEP=1 go test -timeout 300s -run ^TestSweepAll ./pkg/sdk -v
+cleanup: ## cleanup development dependencies
+	rm -rf bin/*
+.PHONY: cleanup
 
-lint:  ## run the fast go linters
-	./bin/reviewdog -conf .reviewdog.yml  -diff "git diff main"
-.PHONY: lint
+sweep: ## destroy architecture; USE ONLY FOR DEVELOPMENT ACCOUNTS
+	@echo "WARNING: This will destroy infrastructure. Use only in development accounts."
+	@read -p "Are you sure? [y/n]" -n 1 REPLY; echo; \
+		if [[ $$REPLY =~ ^[yY]$$ ]]; then \
+		  SNOWFLAKE_ENABLE_SWEEP=1 go test -timeout 300s -run ^TestSweepAll ./pkg/sdk -v; \
+		  else echo "Aborting..."; \
+		fi;
+.PHONY: sweep
 
 lint-ci: ## run the fast go linters
 	./bin/reviewdog -conf .reviewdog.yml -reporter=github-pr-review -tee -fail-on-error=true
 .PHONY: lint-ci
 
-lint-all:  ## run the fast go linters
-	./bin/reviewdog -conf .reviewdog.yml  -filter-mode nofilter
-.PHONY: lint-all
-
-lint-missing-acceptance-tests:
-	@for r in `ls pkg/resources/ | grep -v list_expansion | grep -v privileges | grep -v grant_helpers | grep -v test | xargs -I{} basename {} .go`; do \
-		if [ ! -f pkg/resources/"$$r"_acceptance_test.go ]; then \
-			echo $$r; \
-		fi; \
-	done
-.PHONY: lint-missing-acceptance-tests
-
-build: ## build the binary
-	go build ${LDFLAGS} -o $(BASE_BINARY_NAME) .
-.PHONY: build
-
-coverage: ## run the go coverage tool, reading file coverage.out
-	go tool cover -html=coverage.txt
-.PHONY: coverage
-
-test:  ## run the tests (except sdk tests)
-	CGO_ENABLED=1 $(go_test) -race -coverprofile=coverage.txt -covermode=atomic $(TESTARGS) ./pkg/resources/...
-	CGO_ENABLED=1 $(go_test) -race -coverprofile=coverage.txt -covermode=atomic $(TESTARGS) ./pkg/provider/...
-	CGO_ENABLED=1 $(go_test) -race -coverprofile=coverage.txt -covermode=atomic $(TESTARGS) ./pkg/snowflake/...
-
+test: ## run the tests (except sdk tests)
+	CGO_ENABLED=1 go test -race $(COVERAGE_FLAGS) ./pkg/resources/...
+	CGO_ENABLED=1 go test -race $(COVERAGE_FLAGS) ./pkg/provider/...
+	CGO_ENABLED=1 go test -race $(COVERAGE_FLAGS) ./pkg/snowflake/...
 .PHONY: test
 
 test-acceptance: ## runs all tests, including the acceptance tests which create and destroys real resources
-	SKIP_MANAGED_ACCOUNT_TEST=1 SKIP_EMAIL_INTEGRATION_TESTS=1 TF_ACC=1 $(go_test) -timeout 1200s -v -coverprofile=coverage.txt -covermode=atomic $(TESTARGS) ./...
+	SKIP_MANAGED_ACCOUNT_TEST=1 SKIP_EMAIL_INTEGRATION_TESTS=1 TF_ACC=1 go test -timeout 1200s -v $(COVERAGE_FLAGS) ./...
 .PHONY: test-acceptance
 
-deps:
-	go mod tidy -compat=1.19
-.PHONY: deps
-
-install: ## install the terraform-provider-snowflake binary in $GOPATH/bin
-	go install ${LDFLAGS} .
-.PHONY: install
+build-local: ## build the binary locally
+	go build -o $(BASE_BINARY_NAME) .
```

Comment:
I think it should depend on usage. I mean: do the person building plugin locally care about the tests: if they compile? if they pass? Depending on that we can have 3 different dependencies:
- none (as is, we just want to build)
- dependency to check-fmt (fmt-check) which runs golangci-lint which checks also if tests compile
- dependency to test/accepatance-test/some new rule with tests - which would check if the tests succeed before building

Personally I prefer first option, but I can also accept second one (so the suggested dependency of golangci-lint run).

Reply:
You're right, I guess sometimes you might want to only get the binary and you don't care if the tests compile or not.
However, I think for developers it'd be nice to have a local check to make sure the pipeline won't fail because we forgot to update a test case after renaming a variable which makes it not compile.
So maybe we could add a separate recipe like `compile` which does that and underneath we might just golangci-lint or any other option we decide to use.

Reply:
I was thinking about something similar, like a precommit hook (but not a precommit hook), that checks these mentioned CI aspects. So not only the compilation, but also: were the docs generated? were the dependencies updated? Is formatting good? So basically I would wrap up all *-check rules into one, that we can run before pushing changes. WDYT?

Reply:
After discussion I have added a pre-push rule, which later can be added to pre-push git hook if we all agree on it (or we can run it conditionally)